### PR TITLE
feat(es-dev-server): improve node-resolve performance

### LINKED ISSUE #664: [es-dev-server] npm link not handled by the server
I just linked a component from a local folder using `npm link` command. It can see linked folder in proper location however when I run demo page I see 404 error:
```
GET http://localhost:8080/arc-demo-helper/ApiDemoPage.js net::ERR_ABORTED 404 (Not Found)
```
This is the component I am importing in my demo page. The file is there but it is a link. The setup was working perfectly before making the link so it has to be server.

### PR DESCRIPTION
**Pull request by LarsDenBakker**
https://github.com/open-wc/open-wc/pull/704 was merged, but I had to revert as I didn't cover some scenario's properly. That's fixed now in this PR.

The current implementation of node-resolve uses babel. This is wasteful, as we're creating a full AST of each js file just for resolving the imports.

In this implementation we use https://github.com/guybedford/es-module-lexer. This just extracts information about where imports and dynamic imports are, so that we can resolve them and do a simple string replacement.

Depending on the project, this can significantly speed up performance. Running the dev server and importing all of lodash-es takes 5200ms to load on master, and it takes 1800ms in this branch. Running large bundled libraries like sinon, chai, axe etc. through babel take much longer so that's a huge boost there.

On my machine, running the open-wc tests on master takes 5560ms while on this branch it takes 3670ms. This is without coverage turned on, because that still boots up babel. I also updated karma-esm to only run babel on files that need coverage. That brings it down to 4400ms with coverage enabled.

Fixes #664
Fixes https://github.com/open-wc/open-wc/issues/739

### CODE REVIEW

File: packages/es-dev-server/test/middleware/compile.test.js [hunk 1]
```diff
@@ -141,14 +69,14 @@ describe('babel middleware', () => {
 
       try {
         const responseHtml = await fetch(`${host}src/fake-module.html`);
-        const responseFoo = await fetch(`${host}src/fake-module.foo`);
+        // const responseFoo = await fetch(`${host}src/fake-module.foo`);
```
Comment (daKmoR) [inline packages/es-dev-server/test/middleware/compile.test.js]: commented text?

File: packages/es-dev-server/test/start-server.test.js [hunk 2]
```diff
@@ -1,121 +1,121 @@
...
-});
+// /* eslint-disable no-restricted-syntax, no-await-in-loop */
+// import { expect } from 'chai';
+// import fetch from 'node-fetch';
+// import path from 'path';
+// import { startServer, createConfig } from '../src/es-dev-server.js';
+
+// const host = 'http://localhost:8080/';
+
+// describe('server', () => {
```
Comment (daKmoR) [inline packages/es-dev-server/test/start-server.test.js]: full test disabled?

Comment (LarsDenBakker) [reply packages/es-dev-server/test/start-server.test.js]: Missed this one, thanks!

File: packages/es-dev-server/test/middleware/compile.test.js (hunk 1 above)
Comment (LarsDenBakker) [reply packages/es-dev-server/test/middleware/compile.test.js]: Missed this one, thanks!

**Pull request merged**